### ISSUE TITLE
Interactivity during build steps

### ISSUE THREAD

**Issue by calvinytong** [tilt-dev/tilt#3830]
We have a custom_build and local_resource that call out to git fetch. These builds cause Tilt to hang if the user has cloned in https or their ssh key has a passcode because it will wait forever for user input. It would be nice to be able to allow users to respond to these prompts at build time.

**Comment by joseph-zhong** [comment]
This reminds me of earlier discussion with @nicks about possible future deprecation for the native Terminal/shell interface in favor of interfacing via the browser, which also seems like a good direction here?

**Comment by nicks** [comment]
Hi @calvinytong @joseph-zhong ! 

as of Tilt v0.17.5, we should be setting GIT_TERMINAL_PROMPT=0 in the environment of all custom_build and local_resource commands. I think that should prevent passcode prompts...are you still seeing it wait forever after you upgrade?

Background: I spent some time researching how the `go` team handled this for `go get`. (`go get` also uses `git` underneath to fetch code). This is how they suppress it. I could imagine supporting stdin interfacing via the browser (like jupyter does) for other reasons, but I go back and forth on if that would make sense for password entry.

**Comment by calvinytong** [comment]
Yeah I think that's working correctly, but the trouble is that setting that we'd like to support users with different git configurations. Right now we can only support users who are cloned in ssh and have not password protected their ssh keys. We'd like to be able to support whatever local configuration the user has. 

I'm fine triaging this to low priority if it's not on any current roadmaps since we have workarounds now
